adsmp/solr_updater.py

```diff
@@ -34,6 +34,20 @@ def extract_data_pipeline(data, solrdoc):
         grant.append(grant_no)
         grant_facet_hier.extend(generate_hier_facet(agency, grant_no))
 
+    gpn = []
+    gpn_ids = []
+    gpn_facet_hier_2level = []
+    gpn_facet_hier_3level = []
+    for x in data.get("gpn", []):
+        planet, feature, feature_name, gpn_id = x.split("/", 3)
+        gpn.append(planet)
+        gpn.append(feature)
+        gpn.append(feature_name)
```

Comment:
This isn't going to quite give you what you're looking for - the output here should be (for the two examples you give in your tests - pretend they belong to the same bibcode): `[Moon/Crater/Langrenus, Mars/Terra/Terra Cimmeria]`. But what this will actually give you is: `[Moon, Crater, Langrenus, Mars, Terra, Terra Cimmeria]`. 

I'm guessing you'll actually want something along the lines of: 
`gpn.append("/".join[planet, feature, feature_name])`
but double check that.